api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionTimeLagService.java

```diff
@@ -160,14 +180,69 @@ private Duration getNextEventTimeLag(final NakadiCursor cursor) throws ErrorGett
                 } else {
                     return Duration.ofMillis(new Date().getTime() - nextEvent.getTimestamp());
                 }
-            } catch (final IOException e) {
-                throw new InconsistentStateException("Unexpected error happened when getting consumer time lag", e);
             } catch (final InvalidCursorException e) {
                 throw new ErrorGettingCursorTimeLagException(cursor, e);
             } finally {
                 LOG.trace("client:{}, finished reading events for lag calculation", clientId);
+                consumerPool.returnConsumer(consumer);
+            }
+        }
+    }
+
+    private static class ConsumerPool {
+        private final BlockingQueue<HighLevelConsumer> pool;
+        private final Meter consumerPoolTakeMeter;
+        private final Meter consumerPoolReturnMeter;
+
+        ConsumerPool(final int consumerPoolSize,
+                     final Supplier<HighLevelConsumer> consumerCreator,
+                     final MetricRegistry metricsRegistry) {
+            LOG.info("Preparing timelag checker pool of {} Kafka consumers", consumerPoolSize);
+            this.pool = new LinkedBlockingQueue(consumerPoolSize);
+            for (int i = 0; i < consumerPoolSize; ++i) {
+                this.pool.add(consumerCreator.get());
+            }
+
+            this.consumerPoolTakeMeter = metricsRegistry.meter("nakadi.kafka.consumer.taken");
+            this.consumerPoolReturnMeter = metricsRegistry.meter("nakadi.kafka.consumer.returned");
```

Comment:
```suggestion
            this.consumerPoolTakeMeter = metricsRegistry.meter("nakadi.consumer.taken");
            this.consumerPoolReturnMeter = metricsRegistry.meter("nakadi.consumer.returned");
```
Because it's not Kafka consumer level anymore.

Reply:
Or "timelag-consumer", for example.